client/post-editor/media-modal/detail/detail-title.jsx

```diff
@@ -69,10 +69,6 @@ export default React.createClass( {
 				title: this.state.title
 			} );
 		}
-
-		if ( this.isMounted() ) {
-			this.resetTitle();
```

Comment:
>was the original intent here to make sure we used the item.title over state.title in getTitleValue()?

It seems that way, though been quite a while since this was implemented. It should be the case that resetting the title simply unsets the state, which at this point _should_ defer to the known saved value of the title. I'd expect that if the previous lines are calling to `update`, shouldn't it be optimistic enough to use this as the base title?

Related: 8402-gh-calypso-pre-oss

From there, a few things to test for regressions:

>A title will only be saved if (a) it's not empty, (b) it's not equal to the previous title, (c) the "Escape" character is not pressed, and (d) the user removes focus from the field or presses "Enter". In all other cases, the title will be reset to its original value.